planner/core/logical_plan_builder.go

```diff
@@ -3903,7 +3903,7 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		columns = tbl.WritableCols()
 	} else if b.inDeleteStmt {
 		// All hidden columns are needed because we need to delete the expression index that consists of hidden columns.
```

Comment:
We need to update this comment.

Reply:
FullHiddenColsAndVisibleCols() is useful and depended in the case of partion prune, so I keeped it.

```
func (s *partitionProcessor) reconstructTableColNames(ds *DataSource) ([]*types.FieldName, error) {
	names := make([]*types.FieldName, 0, len(ds.TblCols))
	colsInfo := ds.table.FullHiddenColsAndVisibleCols()
	colsInfoMap := make(map[int64]*table.Column, len(colsInfo))
	for _, c := range colsInfo {
		colsInfoMap[c.ID] = c
	}
        ...
```